### PR TITLE
Updated goflags to latest + misc

### PR DESCRIPTION
**Pull request by Ice3man543**
## Proposed changes

Updated goflags to latest version

## Checklist

- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [ ] All checks passed (lint, unit/integration/regression tests etc.) with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by ehsandeep** [comment]
Depends on https://github.com/projectdiscovery/goflags/issues/73

File: v2/cmd/nuclei/main.go [hunk 1]
```diff
@@ -113,40 +113,40 @@ on extensive configurability, massive extensibility and ease of use.`)
 	*/
 
 	flagSet.CreateGroup("input", "Target",
-		flagSet.StringSliceVarP(&options.Targets, "target", "u", []string{}, "target URLs/hosts to scan"),
+		flagSet.StringSliceVarP(&options.Targets, "target", "u", []string{}, "target URLs/hosts to scan", goflags.StringSliceOptions),
```
Comment (Mzack9999) [inline v2/cmd/nuclei/main.go:116]: What if we use `nil` instead of `[]string{}`?

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
Suggesting to use `nil` instead of `[]string{}`

File: v2/cmd/nuclei/main.go (hunk 1 above)
Comment (Ice3man543) [reply v2/cmd/nuclei/main.go:116]: this change is unrelated to this PR and should be done separately

**Pull request merged**